### PR TITLE
Skip compliance and testing steps where possible on branches and tags

### PR DESCRIPTION
**Pull request by lafriks**
As title. Run only unit-test and mysql integration tests on master branch for coverage report

Hopefully I did not break anything, not something I could test :roll_eyes:

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -92,6 +96,13 @@ workspace:
   base: /go
   path: src/code.gitea.io/gitea
 
+trigger:
```
Comment (CirnoT) [inline .drone.yml:202]: Triggers are AND in Drone, so this won't execute for backports

**Review by CirnoT — COMMENTED** [review]
`testing-amd64` depends on `compliance`; I don't remember how Drone will behave if pipeline depends on skipped pipeline.

File: .drone.yml (hunk 1 above)
Comment (lafriks) [reply .drone.yml:202]: you mean backport PRs?

**Comment by lafriks** [comment]
I thought it executes even if dependency is not run, otherwise `notifications` pipline (last one in this file) would not be run ever as it depends on all other pipelines

File: .drone.yml (hunk 1 above)
Comment (CirnoT) [reply .drone.yml:202]: Yes, the condition here is that branch must be `master` and event must be either `pull_request` or `push`.

Comment (lafriks) [reply .drone.yml:202]: yeah, you are right but somehow I can't think of better way without duplicating steps :(

**Comment by lafriks** [comment]
@CirnoT done

**Comment by lafriks** [comment]
I split master push coverage to other pipeline `coverage`

**Comment by CirnoT** [comment]
We already use `coverage` as name for step that uploads to codecov, re-using it for pipeline step is valid but confusing.

File: .drone.yml [hunk 2]
```diff
@@ -219,6 +287,9 @@ steps:
       USE_REPO_TEST_DIR: 1
     depends_on:
       - build
+    when:
+      event:
+        - pull_request
 
   - name: generate-coverage
```
Comment (CirnoT) [inline .drone.yml]: This won't execute on push to master now, so codecov reports will be broken.

**Comment by lafriks** [comment]
@CirnoT done

**Comment by CirnoT** [comment]
Looks good from my end, however someone should run it locally to double verify that dependency graph acts as desired.

**Comment by lafriks** [comment]
Bad bot

**Comment by lunny** [comment]
Please resolve the conflicts

**Comment by zeripath** [comment]
as per discussion on discord - I'm closing this. @lafriks please re-open and fix conflicts if you think we should merge this.

**Pull request closed without merging**